File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if head is None or head.next is None:
            return head

        prev = None
        curr = head
        nxt = head.next
        nxtUp= None
        first = 0

        print(nxt.val)

        while nxt != None:

            if first == 0:
                curr.next = None
                nextUp = nxt.next
                nxt.next = curr
                first = 1
            else:
                curr.next = prev
                nextUp = nxt.next
                nxt.next = curr
            prev = curr
            curr = nxt
            nxt = nextUp

        # curr.next = prev
        return curr



    
        

File: 0206-reverse-linked-list/0206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {

    if(!head){
        return head;
    }

    if(!head.next){
        return head;
    }
    
    let pre = null;
    let curr = head;
    let next = curr.next;

    while(curr.next){
       curr.next = pre;
       pre = curr
       curr = next;
       next = next.next;
    //    next = next.next;
    }
    curr.next = pre

    // console.log(pre)
    return curr;


    
};